k9mail/src/main/java/com/fsck/k9/message/html/HttpUriParser.java

```diff
@@ -79,9 +75,9 @@ public int linkifyUri(String text, int startPos, StringBuffer outputBuffer) {
         return currentPos;
     }
 
-    private int matchUserInfoIfAvailable(String text, int startPos, int authorityEnd) {
+    private int matchUserInfoIfAvailable(String text, int startPos) {
         int userInfoEnd = text.indexOf('@', startPos);
-        if (userInfoEnd != -1 && userInfoEnd < authorityEnd) {
```

Comment:
`authorityEnd` is still a useful upper bound that can be used to avoid useless work.